src/submission-handlers/encounterRoleHandler.ts

```diff
@@ -0,0 +1,28 @@
+import { type EncounterContext, type FormField, type OpenmrsEncounter, type SubmissionHandler } from '..';
+
+export const EncounterRoleHandler: SubmissionHandler = {
+  handleFieldSubmission: (field: FormField, value: any, context: EncounterContext) => {
+    context.setEncounterRole(value);
+    return value;
+  },
+  getInitialValue: (
+    encounter: OpenmrsEncounter,
+    field: FormField,
+    allFormFields: Array<FormField>,
+    context: EncounterContext,
+  ) => {
+    if (encounter) {
+      return encounter.encounterProviders[0]?.encounterRole;
```

Comment:
Can a provider have more than one encounter role?

Reply:
I am not sure but the provider is saved as an array with provider uuid and encounterRole uuid